common/scripts/setup_env.sh

```diff
@@ -98,8 +98,11 @@ fi
 if [[ -d "${HOME}/.config/gcloud" ]]; then
   CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=${HOME}/.config/gcloud,destination=/config/.config/gcloud,readonly "
 fi
-if [[ -d "${HOME}/.kube" ]]; then
-  CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=${HOME}/.kube,destination=/home/.kube "
+if [[ -n "${KUBECONFIG}" ]]; then
+  CONDITIONAL_HOST_MOUNTS+="--mount type=bind,source=$(dirname "${KUBECONFIG}"),destination=/home/.kube,readonly "
```

Comment:
What happens if KUBECONFIG has multiple files?

Reply:
yes iteration happens over kubeconfig entries and are added as read-only randomized file mounts.